mesonbuild/modules/pkgconfig.py

```diff
@@ -328,23 +330,24 @@ def generate_libs_flags(libs):
                 ofile.write('Libs: {}\n'.format(' '.join(generate_libs_flags(deps.pub_libs))))
             if len(deps.priv_libs) > 0:
                 ofile.write('Libs.private: {}\n'.format(' '.join(generate_libs_flags(deps.priv_libs))))
```

Comment:
I think these should also be conditional on `dataonly`, and having any library deps here should be an error. A dataonly pkg-config file can't really pull in libraries, that makes no sense semantically.

Reply:
Yep

Reply:
@nirbheek any thoughts?

Reply:
You're right, this is not needed.